src/m365/pa/commands/app/app-remove.ts

```diff
@@ -0,0 +1,108 @@
+import * as chalk from 'chalk';
+import { Cli, Logger } from '../../../../cli';
+import {
+  CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import Utils from '../../../../Utils';
+import AzmgmtCommand from '../../../base/AzmgmtCommand';
+import commands from '../../commands';
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  name: string;
+  confirm?: boolean;
+}
+
+class PaAppRemoveCommand extends AzmgmtCommand {
+  public get name(): string {
+    return commands.APP_REMOVE;
+  }
+
+  public get description(): string {
+    return 'Removes the specified Microsoft Power App';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.confirm = typeof args.options.confirm !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    if (this.verbose) {
+      logger.logToStderr(`Removing Microsoft Power App ${args.options.name}...`);
+    }
+
+    const removePaApp: () => void = (): void => {
+      const requestOptions: any = {
+        url: `${this.resource}providers/Microsoft.PowerApps/apps/${encodeURIComponent(args.options.name)}?api-version=2017-08-01`,
+        resolveWithFullResponse: true,
+        headers: {
+          accept: 'application/json'
+        },
+        responseType: 'json'
+      };
+
+      request
+        .delete(requestOptions)
+        .then((rawRes: any): void => {
+          // handle 204 and throw error message to cmd when invalid Power App id is passed
+          // https://github.com/pnp/cli-microsoft365/issues/1063#issuecomment-537218957
+          if (rawRes.statusCode === 204) {
```

Comment:
When using the `providers/Microsoft.PowerApps/apps/` API, when you specify an ID of a non-existent app, you get a 403 response with the following error:

```text
The user with object id 'dc0e7738-56ef-4a41-8513-aeb7dcf86cf8' does not have permission to access the path '/providers/Microsoft.PowerApps/apps/9bfdb8fb-32e7-4716-95bf-9f0bcb80f12a'.
```

We should adjust the code so that the returned error message is displayed to the user as-is.

Reply:
Yes, you're right. Working on getting this in. Will let you know when ready.

Reply:
Think I fixed it by just changing the 204 -> 403. The API doesn't seem to be giving 204. Kept the message as you suggested. Do I need to handle the 401 in case a user tries to delete a Power App but has no permissions for that?

Reply:
Nope, no need to handle a 401 specifically. But a 403 will likely result in a rejected promise, rather than a successful one which we have now, so you'd need to double check if that's the case and adjust tests if necessary.

Reply:
Sorry about that. I was rushing it a bit. Now handling the rejected promise, check status and provide corresponding message. I removed the 204 check on promise resolved. Tests are updated and are passing. Code coverage threshold met.